android/apollo/src/com/andrew/apollo/provider/RecentStore.java

```diff
@@ -94,40 +98,79 @@ public static synchronized RecentStore getInstance(final Context context) {
+     * @param artistName The artist song name.
      * @param albumName  The album name.
-     * @param artistName The artist album name.
-     * @param songCount  The number of tracks for the album.
-     * @param albumYear  The year the album was released.
+     * @param duration   The song total time playback.
+     * @param unknownString A default placeholder if some information isn't provided.
      */
-    public void addAlbumId(final Long albumId, final String albumName, final String artistName,
-                           final String songCount, final String albumYear) {
-        if (albumId == null || albumName == null || artistName == null || songCount == null) {
-            return;
+    public void addSongId(final Long songId, final String songName,
+                          final String artistName, final String albumName,
+                          final long duration, String unknownString) {
+
+        if (songId == null || duration == 0) {
+            return ;
         }
+
         try {
             final ContentValues values = new ContentValues(6);
-            values.put(RecentStoreColumns.ID, albumId);
-            values.put(RecentStoreColumns.ALBUMNAME, albumName);
-            values.put(RecentStoreColumns.ARTISTNAME, artistName);
-            values.put(RecentStoreColumns.ALBUMSONGCOUNT, songCount);
-            values.put(RecentStoreColumns.ALBUMYEAR, albumYear);
-            values.put(RecentStoreColumns.TIMEPLAYED, System.currentTimeMillis());
+            values.put(BaseColumns._ID, songId);
+            values.put(AudioColumns.TITLE, songName != null ? songName : unknownString);
+            values.put(AudioColumns.ARTIST, artistName != null ? artistName: unknownString);
+            values.put(AudioColumns.ALBUM, albumName != null ? albumName: unknownString);
+            values.put(AudioColumns.DURATION, duration);
+            values.put(RecentStoreColumns.LAST_TIME_PLAYED, System.currentTimeMillis());
+
             writableDatabase.beginTransaction();
-            writableDatabase.delete(RecentStoreColumns.NAME,
-                    RecentStoreColumns.ID + " = ?",
-                    new String[]{
-                            String.valueOf(albumId)
-                    });
-            writableDatabase.insert(RecentStoreColumns.NAME, null, values);
+            writableDatabase.delete(RecentStoreColumns.TABLE_NAME,
+                    BaseColumns._ID + " = ?", new String[]{ String.valueOf(songId) });
+            writableDatabase.insert(RecentStoreColumns.TABLE_NAME, null, values);
             writableDatabase.setTransactionSuccessful();
             writableDatabase.endTransaction();
         } catch (Throwable e) {
-            // not critical at all
             e.printStackTrace();
         }
     }
 
+    /**
+     * Used to retrieve the most recently listened song for an artist.
+     *
+     * @param key The key to reference.
+     * @return The most recently listened song for an artist.
```

Comment:
I think you meant:
"@return The most recently listened song name for an artist."